Sources/TuistKit/ProjectEditor/ProjectEditor.swift

```diff
@@ -120,4 +144,61 @@ final class ProjectEditor: ProjectEditing {
         try writer.write(workspace: descriptor)
         return descriptor.xcworkspacePath
     }
+
+    /// Attempts to load the plugins at the given path. If unable to find the Config manifest
+    /// displays a warning to the user.
+    /// - Returns: The loaded `Plugins`.
+    private func loadPlugins(at path: AbsolutePath) -> Plugins {
+        do {
+            let config = try configLoader.loadConfig(path: path)
+            return try pluginService.loadPlugins(using: config)
+        } catch {
+            logger.warning("Failed to load plugins, attempt to fix the Config manifest and rerun the command. Continuing...")
+            logger.debug("Failed to load plugins during edit: \(error.localizedDescription)")
+            return .none
+        }
+    }
+
+    /// - Returns: A list of manifest name and path for plugin manifests that are editable and should be
+    /// loaded as a target in the generated project.
+    private func locateEditablePluginManifests(at path: AbsolutePath, plugins: Plugins) -> [(String, AbsolutePath)] {
+        let editingPluginManifests = manifestFilesLocator.locatePluginManifests(at: path)
+        let loadedHelpers = plugins.projectDescriptionHelpers
+
+        // If a loaded plugin is also a locally editable plugin, we should take the name of the loaded plugin (defined in Plugin.swift manifest)
+        // Otherwise, use the name of the parent directory as the name of the plugin.
+        return editingPluginManifests.map { editableManifest in
+            if let loadedEditableManifest = loadedHelpers.first(where: { $0.path.parentDirectory == editableManifest.parentDirectory }) {
+                return (loadedEditableManifest.name, editableManifest)
+            } else {
+                return (editableManifest.parentDirectory.basename, editableManifest)
+            }
+        }
+    }
+
+    /// Attempts to build the loaded plugins. If it fails, shows a warning to the user.
+    /// - Returns: The built plugin helper modules.
+    private func buildPluginModules(
```

Comment:
I think I need some context here. Why are we building the plugins for editing the project?

Reply:
Sure! So for context we need to build and link plugins that are remote or that don't exist locally.

Let's break it down further:
- If its a remote plugin, we first fetch it from git. From here we checkout to a tag or branch. We don't want to allow users to edit these files since they're in a cache directory and may be wiped at any time. To stop this we instead build the remote plugin and link it so that users can import it in their project.
- If its a local plugin that exists outside of the directory where `tuist edit` was called, then this plugin wont be loaded as part of the xcode project we create for editing plugins (since we only grab plugins that are found underneath the current edit directory). In this case we take this local plugin and build it and link it as we do with the remote one. Again, this use case is only for local plugins that are in some other directory on the machine. For example:
  ```
  - root
  -- MyPlugin
       -- Plugin.swift
  -- MyProject
      -- Project.swift
  ```
  If we ran `tuist edit` from inside `MyProject` then we build and link `MyPlugin`, if instead we ran `tuist edit` from within `root` wed have `MyPlugin` as an editable target in the generated edit workspace so we can instead include it as a target (which means you can modify the sources, build, etc)

Reply:
> If its a local plugin that exists outside of the directory where tuist edit was called, then this plugin wont be loaded as part of the xcode project we create for editing plugins (since we only grab plugins that are found underneath the current edit directory).

I wonder if it would maybe make sense to load local plugins into the project.

The use case I am thinking about is the following:
- I have a project with some helpers and I want to extract them to a plugin. To do that I'd appreciate if I could simultaneously edit the plugin and the project I am writing the plugins for.

To me this is the same as for remote x local Swift packages. You are not able to edit remote packages but you are able to edit the local ones.

Also, if I understand it correctly, we want to pre-build remote plugins because their sources location is something we don't want to expose - which is not the case for local plugins, though.

Would it make sense to you @luispadron to change this and make local plugins, even if they are in a different directory, editable and not pre-build them?

Reply:
Yeah absolutely! I mostly didn't implement it originally because I thought it would be more complex but I have been able to [simplify the logic](https://github.com/tuist/tuist/pull/2642/files#diff-d946a3a89f2e016cbff04216979640f9404403f36ce454184a4442fa4a1e3a68R145-R155).

So now we load both local plugins to the `tuist edit` command as well as any plugins that is on the current file system into the current `Plugin.xcodeproj` as a target.

**One difference still remains**: When editing with a broken config manifest that is in the directory/subdirectory of where the `tuist edit` command was run we are able to still load the plugin as an editable target in the project, the same is not true for local plugins outside of this execution path, since we aren't able to determine where they are located.